doc/source/trajectories.rst

```diff
@@ -1,11 +1,157 @@
 .. -*- coding: utf-8 -*-
 
-===============
+============
 Trajectories
-===============
+============
 
-MDAnalysis groups dynamic data about a :code:`Universe` into its trajectory. This is typically loaded from a trajectory file.
+In MDAnalysis, static data is contained in your universe Topology, while dynamic data is drawn from its trajectory at ``Universe.trajectory``. This is typically loaded from a trajectory file and includes information such as:
 
-Trajectory transformations
+    * atom coordinates (``Universe.atoms.positions``)
+    * box size (``Universe.dimensions``)
+    * velocities and forces (if your file format contains the data) (``Universe.atoms.velocities``)
+
+Although these properties look static, they are actually dynamic, and the data contained within can change.
+In order to remain memory-efficient, MDAnalysis does not load every frame of your trajectory into memory at once. Instead, a Universe has a state: the particular timestep that it is currently associated with in the trajectory. When the timestep changes, the data in the properties above shifts accordingly.
+
+The typical way to change a timestep is to index it. ``Universe.trajectory`` can be thought of as a list of :class:`~MDAnalysis.coordinates.base.Timestep`\ s, a data structure that holds information for the current time frame. For example, you can query its length.
+
+.. ipython:: python
+
+    import MDAnalysis as mda
+    from MDAnalysis.tests.datafiles import PSF, DCD
+
+    u = mda.Universe(PSF, DCD)
+    len(u.trajectory)
+
+When a trajectory is first loaded from a file, it is set to the first frame by default.
+
+.. ipython:: python
+
+    print(u.trajectory.ts, u.trajectory.time)
+
+Indexing the trajectory returns the timestep for that frame, and sets the Universe to point to that frame until the timestep next changes.
+
+.. ipython:: python
+
+    u.trajectory[3]
+
+.. ipython:: python
+
+    print('Time of fourth frame', u.trajectory.time)
+
+Many tasks involve applying a function to each frame of a trajectory. For these, you need to iterate through the frames, *even if you don't directly use the timestep*. This is because the act of iterating moves the Universe onto the next frame, changing the dynamic atom coordinates. 
+
+Trajectories can be sliced as below, if you only want to work on a subset of frames.
```

Comment:
There could be a short subpage on trajectory slicing — things like boolean indexing and fancy indexing also work and provide a powerful way to work with data.

Furthermore, sliced trajectory iterators ([FrameIterators](https://www.mdanalysis.org/mdanalysis/documentation_pages/coordinates/base.html#frameiterators)) allow one to pre-slice a trajectory and then hand this pre-sliced trajectory object over to other code that can use it as if it were a normal trajectory (See also https://www.mdanalysis.org/2018/12/01/release-0.19.x/#improvements-to-file-readers)  

One example (somewhat contrived when applied to our example data)
1. find all frames where observable such as RMSD is within a certain range
2. index trajectory with frame array or boolean array
3. use the sliced trajectory for a calculation such as an average structure

(Sadly, one cannot pass a FrameIterator to one of our analysis functions — this would be very neat for custom analysis on arbitrary subsets of trajectories.)

Reply:
Thank you, I wasn’t aware of that functionality and added a new section. Could AnalysisBase have a function to slice the trajectory before calling run()?